src/transformer.ts

```diff
@@ -1,193 +1,581 @@
-import crypto from 'crypto';
-import libxslt from 'libxslt';
-import type {
-    Document as XMLJSDocument,
-    DocumentFragment as XMLJSDocumentFragment,
-} from 'libxmljs';
-import pkg from '../package.json';
-import xslForm from './xsl/openrosa2html5form.xsl?raw';
-import xslModel from './xsl/openrosa2xmlmodel.xsl?raw';
+import { parseHTML, parseXML, serializeHTML, serializeXML } from './dom';
 import { parseLanguage } from './language';
```

Comment:
General note on the changes in `transformer.ts` (the bulk of the substantive implementation changes): for the most part, I've tried to preserve existing naming, semantics, and logical flow. That includes functions which were already defined, as well as portions of the XSLT which have been reimplemented here.

For those portions of XSLT, I've also tried to preserve, clarify, and expand upon any comments that were already present, as well as add as much commentary of my own as I felt could be helpful understanding what and why each thing is.

And at least for review purposes, I wanted to be sure that each _portion_ of XSLT reimplemented here has its own commit. Following review, I suspect we'll want to squash a large part of the commit history, but I think it's worth considering preserving those commits specifically, as they'll almost certainly be helpful context for the future.